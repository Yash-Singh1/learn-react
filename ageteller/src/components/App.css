#App {
  padding: 5%;
  text-align: center;
  font-size: 16px;
}

#submit {
  margin-top: 5px;
}

#party-popper {
  width: 200px;
  height: 200px;
}

.fade {
  animation-name: fade-in;
  animation-duration: 1s;
}

.fade-out-in {
  animation-name: fade-out-in;
  animation-duration: 2s;
}

.age-stats {
  opacity: 100
}

@keyframes fade-in {
  from { opacity: 0 }
  to { opacity: 100 }
}

@keyframes fade-out-in {
  0% { opacity: 100 }
  50% { opacity: 0 }
  100% { opacity: 100 }
}
