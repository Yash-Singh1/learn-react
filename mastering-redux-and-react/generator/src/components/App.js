import React, { Component } from 'react';
import { connect } from 'react-redux';
import MemeItem from './MemeItem';
import MyMemes from './MyMemes';
import { Form, FormGroup, FormControl, FormLabel } from 'react-bootstrap';

class App extends Component {
  constructor() {
    super();

    this.state = {
      memeLimit: 10,
      text0: '',
      text1: ''
    };
  }

  render() {
    return (
      <div>
        <h2>
          <u>Welcome to the Meme Generator!</u>
        </h2>
        <MyMemes />
        <h4>
          <i>Write Some Text</i>
        </h4>
        <Form inline>
          <FormGroup>
            <FormLabel>Top</FormLabel>{' '}
            <FormControl
              type="text"
              onChange={(event) => this.setState({ text0: event.target.value })}
            ></FormControl>
          </FormGroup>{' '}
          <FormGroup>
            <FormLabel>Bottom</FormLabel>{' '}
            <FormControl
              type="text"
              onChange={(event) => this.setState({ text1: event.target.value })}
            ></FormControl>
          </FormGroup>
        </Form>
        {this.props.memes.slice(0, this.state.memeLimit).map((meme, index) => {
          return (
            <MemeItem
              key={index}
              meme={meme}
              text0={this.state.text0}
              text1={this.state.text1}
            />
          );
        })}
        {this.state.memeLimit === this.props.memes.length ? (
          <div></div>
        ) : (
          <div>
            <br />
            <button
              className="meme-button btn btn-primary"
              onClick={() =>
                this.setState({ memeLimit: this.state.memeLimit + 10 })
              }
            >
              Load 10 more memes...
            </button>
          </div>
        )}
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    memes: state.memes
  };
}

export default connect(mapStateToProps)(App);
