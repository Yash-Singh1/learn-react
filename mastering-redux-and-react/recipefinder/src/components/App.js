import React, { Component } from 'react';
import SearchRecipes from './SearchRecipes';
import RecipeList from './RecipeList';
import RecipeButtons from './RecipeButtons';
import Banner from './Banner';
import '../styles/index.css';

class App extends Component {
  render() {
    return (
      <div>
        <h2>Recipe Finder</h2>
        <SearchRecipes />
        <RecipeList />
        <RecipeButtons />
        <Banner />
      </div>
    );
  }
}

export default App;
