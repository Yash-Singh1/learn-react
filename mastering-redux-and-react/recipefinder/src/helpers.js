export function search(query, cb) {
  const url = `/api/?i=${query.ingredients}&q=${query.dish}&p=${query.page}`;
  // console.log('url', url);
  fetch(url)
    .then((response) => response.json())
    .then(cb);
}
