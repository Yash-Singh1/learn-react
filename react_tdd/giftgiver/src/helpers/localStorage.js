export const setItem = (...args) => localStorage.setItem(...args);
