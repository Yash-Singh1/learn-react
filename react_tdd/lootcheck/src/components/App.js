import React from 'react';
import Wallet from './Wallet';
import Loot from './Loot';

const App = () => (
  <div>
    <h2>Loot Check</h2>
    <hr />
    <Wallet />
    <hr />
    <Loot />
    <div>
      Powered by{' '}
      <a
        target="_blank"
        rel="noreferrer"
        href="https://www.coindesk.com/price/bitcoin"
      >
        Coindesk
      </a>
    </div>
  </div>
);

export default App;
