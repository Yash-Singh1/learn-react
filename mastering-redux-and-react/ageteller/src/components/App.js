import React, { Component } from 'react';
import { Form, FormControl, Button } from 'react-bootstrap';
import './App.css';
import AgeStats from './AgeStats';

class App extends Component {
  constructor() {
    super();

    this.state = {
      newDate: '',
      birthday: '2007-12-10',
      showStats: 0,
      withoutClass: false
    }
  }

  retriggerAnimation(cb) {
    this.setState({ withoutClass: true });
    setTimeout(() => {
      this.setState({ withoutClass: false });
      setTimeout(cb, 1000);
    }, 100);
  }

  changeBirthday() {
    // console.log(this.state);
    if (isNaN(new Date(this.state.newDate).getDate())) {
      alert('Please enter a valid date');
    } else {
      if (this.state.showStats >= 1) {
        this.setState({
          showStats: this.state.showStats + 1
        });
        this.retriggerAnimation(() => {
          this.setState({
            birthday: this.state.newDate,
          });
        });
      } else {
        this.setState({
          birthday: this.state.newDate,
          showStats: this.state.showStats + 1
        });
      }
    }
  }

  render() {
    return (
      <div id='App'>
        <h2>Input Your Birthday!</h2>
        <Form inline>
          <FormControl
            type="date"
            onChange={event => this.setState({ newDate: event.target.value })}
          >
          </FormControl>
          {' '}
          <Button id='submit' onClick={() => this.changeBirthday()}>
            Submit
          </Button>
        </Form>
        {
          this.state.showStats === 1 ?
            <div className="fade age-stats">
              <AgeStats date={this.state.birthday} />
            </div>
          :
            this.state.showStats > 1 ?
              this.state.withoutClass ?
                <div className="age-stats">
                  <AgeStats date={this.state.birthday} />
                </div>
              :
                <div className="age-stats fade-out-in">
                  <AgeStats date={this.state.birthday} />
                </div>
            :
              <div></div>
        }
      </div>
    )
  }
}

export default App;
