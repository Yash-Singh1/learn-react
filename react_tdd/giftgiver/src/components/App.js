import React, { Component } from 'react';
import { Button } from 'react-bootstrap';
import Gift from './Gift';
import { setItem } from '../helpers/localStorage';

class App extends Component {
  constructor() {
    super();

    /* istanbul ignore else */
    if (!localStorage.getItem('gifts')) {
      localStorage.setItem('gifts', '[]');
    }
    this.state = {
      gifts: JSON.parse(localStorage.getItem('gifts'))
    };
  }

  addGift() {
    this.state.gifts.push({
      id:
        this.state.gifts.length === 0
          ? 1
          : this.state.gifts[this.state.gifts.length - 1].id + 1,
      person: '',
      present: ''
    });
    this.setState({ gifts: this.state.gifts });
  }

  modifyGift(id, data) {
    const gifts = this.state.gifts.map((gift) =>
      gift.id === id ? { ...data, id } : gift
    );
    this.setState({ gifts });
  }

  removeGift(id) {
    const gifts = this.state.gifts.filter((gift) => gift.id !== id);
    this.setState({ gifts });
  }

  componentDidUpdate() {
    setItem('gifts', JSON.stringify(this.state.gifts));
  }

  render() {
    return (
      <div>
        <h2>Gift Giver</h2>
        <div className="gift-list">
          {this.state.gifts.map((gift) => {
            return (
              <Gift
                key={gift.id}
                gift={gift}
                person={gift.person}
                present={gift.present}
                removeGift={(id) => this.removeGift(id)}
                modifyGift={(id, data) => this.modifyGift(id, data)}
              />
            );
          })}
        </div>
        <Button className="btn-add" onClick={() => this.addGift()}>
          Add Gift
        </Button>
      </div>
    );
  }
}

export default App;
