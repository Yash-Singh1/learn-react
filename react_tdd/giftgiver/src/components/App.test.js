import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import * as localstorage from '../helpers/localStorage';

describe('App', () => {
  const app = shallow(<App />);
  const localStorageSetItemSpy = jest.spyOn(localstorage, 'setItem');

  it('renders properly', () => {
    expect(app).toMatchSnapshot();
  });

  it('initializes the `state` with an empty list of gifts', () => {
    expect(app.state().gifts).toEqual([]);
  });

  describe('when clicking the `add-gift` button', () => {
    beforeEach(() => {
      app.find('.btn-add').simulate('click');
    });

    afterEach(() => {
      app.setState({ gifts: [] });
    });

    it('adds a new gift to `state`', () => {
      expect(app.state().gifts).toEqual([{ id: 1, person: '', present: '' }]);
    });

    it('adds a new gift to the rendered list', () => {
      expect(app.find('.gift-list').children().length).toEqual(1);
    });

    it('creates a Gift component', () => {
      expect(app.find('Gift').exists()).toBe(true);
    });

    describe('and the user wants to remove the gift (helper)', () => {
      beforeEach(() => {
        app.instance().removeGift(1);
      });

      it('removes the gift from `state`', () => {
        expect(app.state().gifts).toEqual([]);
      });

      it("doesn't render the gift", () => {
        expect(app.find('.gift-list').children().length).toEqual(0);
      });
    });

    describe('and the user wants to remove the gift (UI)', () => {
      beforeEach(() => {
        app.find('Gift').dive().find('Button.btn-remove').simulate('click');
      });

      it('removes the gift from `state`', () => {
        expect(app.state().gifts).toEqual([]);
      });

      it("doesn't render the gift", () => {
        expect(app.find('.gift-list').children().length).toEqual(0);
      });
    });

    describe('and the user wants to modify the gift', () => {
      const info = { person: 'Uncle', present: 'whatever he wants' };

      beforeEach(() => {
        app.instance().modifyGift(1, info);
      });

      it('modifies the `state`', () => {
        expect(app.state().gifts).toEqual([{ id: 1, ...info }]);
      });

      it('modifies the local storage', () => {
        app.instance().componentDidUpdate();
        expect(localStorageSetItemSpy).toHaveBeenCalledWith(
          'gifts',
          JSON.stringify([{ ...info, id: 1 }])
        );
      });

      it("makes an empty array when the local storage isn't there", () => {
        localStorage.clear();
        new App();
        expect(localStorage.getItem('gifts')).toEqual('[]');
      });

      describe('and the user wants to modify the gift in the UI', () => {
        beforeEach(() => {
          app
            .find('Gift')
            .dive()
            .find('.input-person')
            .simulate('change', { target: { value: 'Uncle2' } });
        });

        it('modifies the gift in state', () => {
          expect(app.state().gifts[0].person).toEqual('Uncle2');
        });
      });
    });

    describe('and the user wants to add another gift', () => {
      beforeEach(() => {
        app.find('.btn-add').simulate('click');
      });

      it('modifies the `state`', () => {
        expect(app.state().gifts).toEqual([
          { id: 1, person: '', present: '' },
          { id: 2, person: '', present: '' }
        ]);
      });

      it('renders the new gift', () => {
        expect(app.find('.gift-list').children().length).toEqual(2);
      });

      describe('and the user wants to modify the second gift', () => {
        const info = { person: 'Uncle', present: 'whatever he wants' };

        beforeEach(() => {
          app.instance().modifyGift(2, info);
        });

        it('modifies the `state`', () => {
          expect(app.state().gifts).toEqual([
            { id: 1, person: '', present: '' },
            { id: 2, ...info }
          ]);
        });
      });
    });
  });
});
