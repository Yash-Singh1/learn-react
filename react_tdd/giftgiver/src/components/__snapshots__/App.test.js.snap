// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
<div>
  <h2>
    Gift Giver
  </h2>
  <div
    className="gift-list"
  />
  <Button
    active={false}
    className="btn-add"
    disabled={false}
    onClick={[Function]}
    variant="primary"
  >
    Add Gift
  </Button>
</div>
`;
